ui/src/app/dashboard/views/DashboardConfigurationForm/DashboardConfigurationSubnetForm/DashboardConfigurationSubnetForm.test.tsx

```diff
@@ -0,0 +1,120 @@
+import { mount } from "enzyme";
+import { Provider } from "react-redux";
+import configureStore from "redux-mock-store";
+
+import DashboardConfigurationSubnetForm from "./DashboardConfigurationSubnetForm";
+
+import { actions as subnetActions } from "app/store/subnet";
+import {
+  fabric as fabricFactory,
+  fabricState as fabricStateFactory,
+  rootState as rootStateFactory,
+  subnet as subnetFactory,
+  subnetState as subnetStateFactory,
+} from "testing/factories";
+
+const mockStore = configureStore();
+
+describe("DashboardConfigurationSubnetForm", () => {
+  it("displays a spinner if subnets have not loaded", () => {
+    const state = rootStateFactory({
+      subnet: subnetStateFactory({ loaded: false }),
+    });
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <DashboardConfigurationSubnetForm />
+      </Provider>
+    );
+
+    expect(wrapper.find("Spinner").exists()).toBe(true);
+  });
+
+  it("displays a spinner if fabrics have not loaded", () => {
+    const state = rootStateFactory({
+      fabric: fabricStateFactory({ loaded: false }),
+    });
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <DashboardConfigurationSubnetForm />
+      </Provider>
+    );
+
+    expect(wrapper.find("Spinner").exists()).toBe(true);
+  });
+
+  it("renders the form if fabrics and subnets have loaded", () => {
+    const state = rootStateFactory({
+      fabric: fabricStateFactory({ loaded: true }),
+      subnet: subnetStateFactory({ loaded: true }),
+    });
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <DashboardConfigurationSubnetForm />
+      </Provider>
+    );
+
+    expect(wrapper.find("FormikForm").exists()).toBe(true);
+  });
+
+  it("displays links for the subnet and its fabric", () => {
+    const subnet = subnetFactory({ id: 1, active_discovery: true, vlan: 2 });
+    const fabric = fabricFactory({ id: 3, vlan_ids: [2] });
+    const state = rootStateFactory({
+      fabric: fabricStateFactory({ items: [fabric], loaded: true }),
+      subnet: subnetStateFactory({ items: [subnet], loaded: true }),
+    });
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <DashboardConfigurationSubnetForm />
+      </Provider>
+    );
+
+    expect(wrapper.find("a[data-test='subnet-link']").prop("href")).toBe(
+      "/MAAS/l/subnet/1"
+    );
+    expect(wrapper.find("a[data-test='fabric-link']").prop("href")).toBe(
+      "/MAAS/l/fabric/3"
+    );
+  });
+
+  it("dispatches actions to update subnet active discovery if they have changed", () => {
```

Comment:
@petesfrench I've added some extra tests to the form, but this one in particular is important. We're testing that, given the Redux store is set up in some way and given that the form is submitted with these values, these Redux actions should have been dispatched. It's a bit cumbersome to have to set up the store for these kinds of tests but it's necessary for the whole React "UI is a function of state" thing.